aem-authoring-toolkit-samples/toolkit-samples-bundle/src/main/java/com/exadel/aem/toolkit/samples/models/WarriorDescriptionComponent.java

```diff
@@ -0,0 +1,263 @@
+@DependsOnTab(tabTitle = WarriorDescriptionComponent.TAB_FILMS, query = "@isLikeFilms")
+@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class WarriorDescriptionComponent {
+
+    public static final String TAB_MAIN = "Main info";
+    public static final String TAB_TASTES = "Tastes";
+    public static final String TAB_FRUITS = "Fruits";
+    public static final String TAB_FILMS = "Films";
+    public static final String EXTERNAL_STYLE_PATH = "/apps/authoring-toolkit/samples/components/content/warrior-description-component/clientlib/styles/rte-style.css";
+
+    @Inject
+    private ResourceResolver resourceResolver;
+
+    @Self
+    private Resource resource;
+
+    @PlaceOnTab(WarriorDescriptionComponent.TAB_MAIN)
+    @RichTextEditor(
+            features = {
+                    RteFeatures.UNDO_UNDO,
+                    RteFeatures.UNDO_REDO,
+                    RteFeatures.SEPARATOR,
+                    RteFeatures.Popovers.EDIT_ALL,
+                    RteFeatures.SEPARATOR,
+                    RteFeatures.Popovers.FORMAT_ALL,
+                    RteFeatures.Popovers.STYLES,
+                    RteFeatures.Popovers.JUSTIFY_ALL,
+                    RteFeatures.SEPARATOR,
+                    RteFeatures.MISCTOOLS_SPECIALCHARS,
+                    RteFeatures.SEPARATOR,
+                    RteFeatures.Popovers.PARAFORMAT
+            },
+            htmlPasteRules = @HtmlPasteRules(
+                    allowImages = false,
+                    allowLists = AllowElement.DISALLOW,
+                    allowTables = AllowElement.DISALLOW
+            ),
+            externalStyleSheets = {WarriorDescriptionComponent.EXTERNAL_STYLE_PATH},
+            styles = {
+                    @Style(cssName = "rte-style", text = "Aggressive style"),
+            },
+            icons = {
+                    @IconMapping(command = "#styles", icon = "textColor"),
+                    @IconMapping(command = "#edit", icon = "textEdit")
+            },
+            htmlLinkRules = @HtmlLinkRules(
+                    targetInternal = LinkTarget.MANUAL,
+                    targetExternal = LinkTarget.BLANK,
+                    protocols = {"http://", "https://"},
+                    defaultProtocol = "https://"
+            ),
+            specialCharacters = {
+                    @Characters(name = "Copyright", entity = "&copy"),
+                    @Characters(name = "Euro sign", entity = "&#x20AC"),
+                    @Characters(name = "Trademark", entity = "&#x2122"),
+                    @Characters(rangeStart = 48, rangeEnd = 70),
+            },
+            formats = {
+                    @ParagraphFormat(tag = "h1", description = "My custom header"),
+                    @ParagraphFormat(tag = "h2", description = "My custom subheader")
+            }
+    )
+    @DialogField(name = "./description")
```

Comment:
We can add InplaceEdditing for that field to show power of @Extend functionality